### PR TITLE
Feature/core 886 improve evan table

### PR DESCRIPTION
**Pull request by datrinh**

### CODE REVIEW

File: core/evancore.vue.libs/src/components/table/table.vue [hunk 1]
```diff
@@ -20,33 +20,50 @@
...
   </b-table>
 </template>
 
 <script lang="ts">
 import Component from './table';
+
 export default Component;
 </script>
 
 <style lang="scss" scoped>
```
Comment (karladler) [inline core/evancore.vue.libs/src/components/table/table.vue]: no *.scss file here 😢

File: core/evancore.vue.libs/src/components/table/table.vue [hunk 2]
```diff
@@ -59,6 +76,7 @@ export default Component;
     &::-webkit-scrollbar-thumb {
       border-radius: 4px;
       background-color: rgba(0, 0, 0, 0.5);
+      box-shadow: 0 0 1px rgba(255, 255, 255, 0.5);
       -webkit-box-shadow: 0 0 1px rgba(255, 255, 255, 0.5);
```
Comment (karladler) [inline core/evancore.vue.libs/src/components/table/table.vue]: not sure how it behaves within `::-webkit-` pseudo selectors, but browser prefixes should actually be added by sass, so `-webkit-box-shadow` may be removed ?

File: core/evancore.vue.libs/src/components/table/table.vue [hunk 3]
```diff
@@ -68,28 +86,83 @@ export default Component;
...
-      height: 64px;
+      height: 56px;
       background-color: white;
+      color: cssVar('gray-600');
       cursor: pointer;
+
       // create spacing between rows without using border-collapse
       // because it causes janky behavior with sticky header
-      border-bottom: 4px solid cssVar('bg-level-3');
+      // border-bottom: 4px solid cssVar('bg-level-3');
```
Comment (karladler) [inline core/evancore.vue.libs/src/components/table/table.vue]: you may remove the uncommented code and also the whole comment block if it's not needed anymore

File: core/evancore.vue.libs/src/components/table/table.vue [hunk 4]
```diff
@@ -68,28 +86,83 @@ export default Component;
...
+
+      &:hover {
+        background-color: cssVar('gray-100');
+        // border creates janky jumping. Using inset shadow as an alternative
+        box-shadow: inset 0px 0px 0px 1px cssVar('gray-200');
```
Comment (karladler) [inline core/evancore.vue.libs/src/components/table/table.vue]: you may avoid the jumpin adding a transparent border in general and just set the color on hover.

File: dapps/assets/src/components/Assets.ts [hunk 5]
```diff
@@ -34,10 +37,24 @@ export default class AssetsComponent extends mixins(EvanComponent) {
...
+
+  @Watch('$route')
+  onRouteChange(): void {
+    this.activeIcon = this.getActiveIcon();
+  }
+
+  private getActiveIcon(): string {
+    const matched = this.$route.matched.map((match) => match.name);
+    const activeRoute = matched.find((route) => typeof route !== 'undefined');
+    return this.navItems.find((item) => item.to.name === activeRoute).icon;
```
Comment (karladler) [inline dapps/assets/src/components/Assets.ts]:
```suggestion

    return this.navItems.find((item) => item.to.name === activeRoute).icon;
```

File: dapps/assets/src/components/Assets.vue [hunk 6]
```diff
@@ -63,3 +69,15 @@ import AssetsComponent from './Assets';
 
 export default AssetsComponent;
 </script>
+
+<style lang="scss" scoped>
+@import '~@evan.network/ui/src/style/utils';
```
Comment (karladler) [inline dapps/assets/src/components/Assets.vue]: no scss file here 🥺

File: dapps/assets/src/components/contacts/Contacts.vue [hunk 7]
```diff
@@ -46,25 +46,25 @@
...
-            icon="mdi mdi-domain"
+            icon="mdi mdi-office-building"
             icon-position="left"
             :class="{ active: filter === 'company' }"
             :label="$t('_assets.contacts.companies')"
             @click="filterByType('company')"
           />
-          <evan-button
-            class="filter-btn ml-3"
+          <!-- <evan-button
```
Comment (karladler) [inline dapps/assets/src/components/contacts/Contacts.vue]: please no uncommented code 🙃

File: core/evancore.vue.libs/src/components/table/table.vue (hunk 3 above)
Comment (datrinh) [reply core/evancore.vue.libs/src/components/table/table.vue]: It's not certain yet, if this implementation meets all our needs. I'd like to keep it around until the testing is done.

File: core/evancore.vue.libs/src/components/table/table.vue (hunk 4 above)
Comment (datrinh) [reply core/evancore.vue.libs/src/components/table/table.vue]: Tried it before, but didn't work. Could spend more time with it, but inset shadows work for now and are supported in all browsers.

**Comment by datrinh** [comment]
> Code looks fine for me. I found some inconsistencies to the wireframes:
> 
> * The life switching asset category icon is a cool feature, but we already have the active state of the category in the sidebar. Should we just include the designed asset icon + `My Asset` text?
> 
> image
> 
> * Asset Sidebar has a scrolling bar everytime.
> 
> image
> 
> * Contacts table is using the perfect full height for displaying the content. Can we also add this style to the digital twins?
>   
>   * contacts:
>     image
>   * twins
>     image
> * contacts creating tons of warnings
>   image

I agree that the dynamic approach to displaying the current asset icon is rather convoluted for only 2 asset categories. I simply followed the concept that we had at that time:
image

After talking to Daniel, I will apply the newest concept and make the icon static.

>contacts creating tons of warnings
This is due to the stupid vue-moment filter not handling undefined inputs. (see here: https://github.com/brockpetrie/vue-moment/issues/75)
This shouldn't matter super much in production AND only happens for "legacy" contacts. New contacts always have a date attached. I would leave it like that and ditch the vue-moment lib in the future.

**Pull request merged**